modules/renter/hostdb/negotiate.go

```diff
@@ -319,3 +311,70 @@ func newRevision(rev types.FileContractRevision, pieceLen uint64, merkleRoot cry
 		NewUnlockHash: rev.NewUnlockHash,
 	}
 }
+
+// Renew negotiates a new contract for data already stored with a host. It
+// returns the ID of the new contract. This is a blocking call that performs
+// network I/O.
+func (hdb *HostDB) Renew(fcid types.FileContractID, newEndHeight types.BlockHeight) (types.FileContractID, error) {
+	hdb.mu.RLock()
+	height := hdb.blockHeight
+	hc, ok := hdb.contracts[fcid]
+	host, eok := hdb.allHosts[hc.IP] // or activeHosts?
+	hdb.mu.RUnlock()
+	if !ok {
+		return types.FileContractID{}, errors.New("no record of that contract")
+	} else if !eok {
+		return types.FileContractID{}, errors.New("no record of that host")
+	} else if newEndHeight < height {
+		return types.FileContractID{}, errors.New("cannot renew below current height")
+	}
+
+	renterCost := host.Price.Mul(types.NewCurrency64(hc.LastRevision.NewFileSize)).Mul(types.NewCurrency64(uint64(newEndHeight - height)))
+	renterCost = renterCost.MulFloat(1.05) // extra buffer to guarantee we won't run out of money during revision
+	payout := renterCost                   // no collateral
+
+	// create file contract
+	fc := types.FileContract{
+		FileSize:       hc.LastRevision.NewFileSize,
+		FileMerkleRoot: hc.LastRevision.NewFileMerkleRoot,
+		WindowStart:    newEndHeight,
+		WindowEnd:      newEndHeight + host.WindowSize,
+		Payout:         payout,
+		UnlockHash:     types.UnlockHash{}, // to be filled in by negotiateContract
+		// TODO: are these correct?
+		RevisionNumber:     0,
+		ValidProofOutputs:  hc.LastRevision.NewValidProofOutputs,
+		MissedProofOutputs: hc.LastRevision.NewMissedProofOutputs,
```

Comment:
RevisionNumber is correct, but you don't want to reuse the outputs from the previous FC otherwise you are going to be double paying for the data.

You want to reset these to what they would be if it were a new file contract, where the host is getting 0 and the renter is getting everything in a refund.